uaclient/api/u/pro/token_info/v1.py

```diff
@@ -0,0 +1,246 @@
+import logging
+from typing import Any, Dict, List
+
+from uaclient import exceptions, util
+from uaclient.api.api import APIEndpoint
+from uaclient.api.data_types import AdditionalInfo
+from uaclient.config import UAConfig
+from uaclient.contract import get_available_resources, get_contract_information
+from uaclient.data_types import (
+    BoolDataValue,
+    DataObject,
+    Field,
+    StringDataValue,
+    data_list,
+)
+from uaclient.entitlements import entitlement_factory
+
+LOG = logging.getLogger(util.replace_top_level_logger_name(__name__))
+
+
+class TokenInfoOptions(DataObject):
+    fields = [
+        Field(
+            "token", StringDataValue, doc="Token to be used for the request"
+        ),
+    ]
+
+    def __init__(self, token: str):
+        self.token = token
+
+
+class AccountInfo(DataObject):
+    fields = [
+        Field("id", StringDataValue),
+        Field("name", StringDataValue),
+    ]
+
+    def __init__(self, id: str, name: str):
+        self.id = id
+        self.name = name
+
+
+class ContractInfo(DataObject):
+    fields = [
+        Field("id", StringDataValue),
+        Field("name", StringDataValue),
+    ]
+
+    def __init__(self, id: str, name: str):
+        self.id = id
+        self.name = name
+
+
+class ServiceInfo(DataObject):
+    fields = [
+        Field("name", StringDataValue),
+        Field("description", StringDataValue),
+        Field("entitled", BoolDataValue),
+        Field("auto_enabled", BoolDataValue),
+        Field("available", BoolDataValue),
+    ]
+
+    def __init__(
+        self,
+        available: bool,
+        description: str,
+        entitled: bool,
+        name: str,
+        auto_enabled: bool,
+    ):
+        self.available = available
+        self.description = description
+        self.entitled = entitled
+        self.name = name
+        self.auto_enabled = auto_enabled
+
+
+class TokenInfoResult(DataObject, AdditionalInfo):
+    fields = [
+        Field("account", AccountInfo),
+        Field("contract", ContractInfo),
+        Field("services", data_list(ServiceInfo)),
+    ]
+
+    def __init__(
+        self,
+        account: AccountInfo,
+        contract: ContractInfo,
+        services: List[ServiceInfo],
+    ):
+        self.account = account
+        self.contract = contract
+        self.services = services
+
+
+def _get_entitlement_information_bool(
```

Comment:
Why do we have the `bool` in the name of the function ?

Reply:
There is currently already a function with the same name in the `status.py` file which returns the information result in strings. I have a better idea to deal with this, maybe we can just have one function instead of two.

Reply:
+1 different function. Given you are not importing the `status.py` one here there is no name conflict, just go with the simplest possible.
It does bother me that a function caled `get_..._bool` returns a `Dict` :|

Reply:
I have changed the function names to be more appropriate to their use cases. something like
`user_facing_entitlement_information` and `application_entitlement_information`

Reply:
updated